lib/template-loader.php

```diff
@@ -104,54 +104,53 @@ function gutenberg_find_template( $template_file ) {
 		)
 	);
 
-	if ( $template_query->have_posts() ) {
-		$template_posts        = $template_query->get_posts();
-		$current_template_post = array_shift( $template_posts );
-
-		// Build map of template slugs to their priority in the current hierarchy.
-		$slug_priorities = array_flip( $slugs );
-
-		// See if there is a theme block template with higher priority than the resolved template post.
-		$higher_priority_block_template_path     = null;
-		$higher_priority_block_template_priority = PHP_INT_MAX;
-		$block_template_files                    = glob( get_stylesheet_directory() . '/block-templates/*.html', 1 );
-		if ( is_child_theme() ) {
-				$block_template_files = array_merge( $block_template_files, glob( get_template_directory() . '/block-templates/*.html', 1 ) );
-		}
-		foreach ( $block_template_files as $path ) {
-			$theme_block_template_priority = $slug_priorities[ basename( $path, '.html' ) ];
-			if (
-				isset( $theme_block_template_priority ) &&
-				$theme_block_template_priority < $higher_priority_block_template_priority &&
-				$theme_block_template_priority < $slug_priorities[ $current_template_post->post_name ]
-			) {
-				$higher_priority_block_template_path     = $path;
-				$higher_priority_block_template_priority = $theme_block_template_priority;
-			}
+	$template_posts        = $template_query->get_posts();
+	$current_template_post = array_shift( $template_posts );
+
+	// Build map of template slugs to their priority in the current hierarchy.
+	$slug_priorities = array_flip( $slugs );
+
+	// See if there is a theme block template with higher priority than the resolved template post.
+	$higher_priority_block_template_path     = null;
+	$higher_priority_block_template_priority = PHP_INT_MAX;
+	$block_template_files                    = glob( get_stylesheet_directory() . '/block-templates/*.html', 1 );
+	if ( is_child_theme() ) {
+			$block_template_files = array_merge( $block_template_files, glob( get_template_directory() . '/block-templates/*.html', 1 ) );
+	}
+	$block_template_files = array_merge( $block_template_files, glob( dirname( __FILE__ ) . '/demo-block-templates/*.html', 1 ) );
```

Comment:
My concern here is that the "demo" fallback shouldn't be bundled in the "template loader". This code will ultimately land in Core and ideally, we should keep the default behavior and the demo enabling/disabling separate.

Reply:
Maybe "demo" is not the right term then.

I imagine Core shipping with a demo/fallback index template like this. So that the site doesn't break when no templates are available and to support template-less themes that only modify styles.

Reply:
Well these will be the default themes IMO, no real need for a built-in fallback.

Reply:
Then can we just remove this line later?

It doesn't hurt, even if the folder is not present.